syntax = "proto3";

option java_package = "kr.uwcj.cs492";

service session {
    rpc login(LoginRequest) returns (SessionResponse) {}
    rpc getUsername(SessionRequest) returns (UsernameResponse) {}
    rpc getFeedList(SessionRequest) returns (FeedListResponse) {}
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message SessionResponse {
    int32 sessionId = 1;
}

message UsernameResponse {
    string username = 1;
}

message FeedListResponse {
    repeated string feedList = 1;
}

message SessionRequest {
    int32 sessionId = 1;
}
